src/Substrait.Core/Relation/Rel.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace Substrait.Relation
+{
+  /// <summary>
+  /// Base type for all relational operators, <see cref="Protobuf.Rel"/>
+  /// </summary>
+  abstract public class Rel
```

Comment:
I wonder if there is benefit in `Rel` having a "Children" property?  Though that could be added later.

Reply:
Nope, this is a fine start.  Feel free to add as you go.  Smaller PRs are easier to review so I generally prefer that and, eventually, I think a minimum size could be "adds something that can be tested or fixes a test/bug" but since we're just getting started I don't think we have to worry about minimum size.

My question was more just curious about your intention and direction.  I think data lineage is something useful in almost all cases (e.g. basic plan debug printing) and so this is cool.